spec/lib/deep_nest_spec.rb

```diff
@@ -1,9 +1,645 @@
+
+        it 'returns true' do
+          is_expected.to be_truthy
+        end
+      end
+
+      describe 'with differing arrays that have deeply nested arrays' do
+        let(:obj1) { [1, [2, [3, 4, [5, 6, [7]]], 8], 9] }
+        let(:obj2) { [1, [2, [3, 4, [5, 6, [7, [10]]]], 8], 9] }
+
+        it 'returns false' do
+          is_expected.to be_falsy
+        end
+      end
+
+      describe 'with matching arrays that have simple hashes' do
+        let(:obj1) { [1, 'hi', 2.0, true, { a: 1, b: 2 }] }
+        let(:obj2) { [1, 'hi', 2.0, true, { a: 1, b: 2 }] }
+
+        it 'returns true' do
+          is_expected.to be_truthy
+        end
+      end
+
+      describe 'with differing arrays that have simple hashes' do
+        let(:obj1) { [1, 'hi', 2.0, true, { a: 1, b: 2 }] }
+        let(:obj2) { [1, 'hi', 2.0, true, { a: 1, b: '2' }] }
+
+        it 'returns false' do
+          is_expected.to be_falsy
+        end
+      end
+
+      describe 'with matching arrays that have deeply nested hashes' do
+        let(:obj1) { [1, 'hi', 2.0, true, hi: [{ a: { b: { c: { d: [{ e: 1 }] } } } }]] }
+        let(:obj2) { [1, 'hi', 2.0, true, hi: [{ a: { b: { c: { d: [{ e: 1 }] } } } }]] }
+
+        it 'returns true' do
+          is_expected.to be_truthy
+        end
+      end
+
+      describe 'with differing arrays that have deeply nested hashes' do
+        let(:obj1) { [1, 'hi', 2.0, true, hi: [{ a: { b: { c: { d: [{ e: 1 }] } } } }]] }
+        let(:obj2) { [1, 'hi', 2.0, true, hi: [{ a: { b: { c: { d: [{ e: 1.0 }] } } } }]] }
+
+        it 'returns false' do
+          is_expected.to be_falsy
+        end
+      end
+    end
+
+    describe 'when parameters are hashes' do
+      describe 'with simple hashes with same key-value pairs and order' do
+        let(:obj1) { { a: 1, b: 'string', c: true, d: 2.2, e: [1, 'a', true, 3.3] } }
+        let(:obj2) { { a: 1, b: 'string', c: true, d: 2.2, e: [1, 'a', true, 3.3] } }
+
+        it 'returns true' do
+          is_expected.to be_truthy
+        end
+      end
+
+      describe 'with simple hashes with same keys but different values' do
+        let(:obj1) { { a: 1, b: 'string', c: true, d: 2.2 } }
+        let(:obj2) { { a: 1, b: 'string', c: false, d: 2.2 } }
+
+        it 'returns false' do
+          is_expected.to be_falsy
+        end
+      end
+
+      describe 'with simple hashes with same values but different keys' do
+        let(:obj1) { { a: 1, b: 'string', c: true, d: 2.2 } }
+        let(:obj2) { { a: 1, b: 'string', 'c': true, e: 2.2 } }
+
+        it 'returns false' do
+          is_expected.to be_falsy
+        end
+      end
+
+      describe 'with simple hashes with same key-value pairs but in different order' do
+        let(:obj1) { { a: 1, b: 'string', c: true, d: 2.2 } }
+        let(:obj2) { { a: 1, b: 'string', d: 2.2, c: true } }
+
+        it 'returns false' do
+          is_expected.to be_falsy
+        end
+      end
```

Comment:
You've done an incredible job of testing this tricky method! This is the only test that needs to change. Ruby hashes *do* track keys in the order they were created, but we want to return true even if two hashes created their keys in a different order. The key order is unimportant for a hash, because it won't affect their actual usage.